### ISSUE TITLE
Redis Client throws NPE if key is not present in Redis while calling mgetMany,mget

### ISSUE THREAD

**Issue by vinscom** [vert-x3/vertx-redis-client#140]
### Version

* vert.x redis 3.7.0:

### Problem
If the key is not present in Redis Server then in response *nil* is returned.
Currently, RedisClientImpl.toJsonArray does not handle *nil* correctly.

### Test
https://github.com/vinscom/vertx-issue-140

### Code and Error
**Sample Code**
```
Observable
            .range(1, 10)
            .flatMapSingle((i) -> client.rxSet("K" + i, "V" + i).andThen(Single.just(i)))
            .map(i -> "K" + i)
            .toList()
            .map(j -> {
              j.add("NK1");
              return j;
            })
            .doOnSuccess(l -> System.out.println("Fetching Keys"  + new JsonArray(l).toString()))
            .flatMap((l) -> client.rxMgetMany(l))
            .subscribe((t) -> testContext.completeNow(), err -> testContext.failNow(err));
```
**Error**
```
java.lang.NullPointerException: null
	at io.vertx.redis.impl.RedisClientImpl.toJsonArray(RedisClientImpl.java:181) ~[vertx-redis-client-3.7.0.jar:3.7.0]
	at io.vertx.redis.impl.RedisClientImpl.lambda$sendJsonArray$5(RedisClientImpl.java:203) ~[vertx-redis-client-3.7.0.jar:3.7.0]
	at io.vertx.redis.client.impl.RedisClient.handle(RedisClient.java:358) ~[vertx-redis-client-3.7.0.jar:3.7.0]
	at io.vertx.redis.client.impl.RESPParser.handleResponse(RESPParser.java:209) ~[vertx-redis-client-3.7.0.jar:3.7.0]
	at io.vertx.redis.client.impl.RESPParser.handleResponse(RESPParser.java:186) ~[vertx-redis-client-3.7.0.jar:3.7.0]
	at io.vertx.redis.client.impl.RESPParser.handle(RESPParser.java:121) ~[vertx-redis-client-3.7.0.jar:3.7.0]
	at io.vertx.redis.client.impl.RESPParser.handle(RESPParser.java:25) ~[vertx-redis-client-3.7.0.jar:3.7.0]
	at io.vertx.core.net.impl.NetSocketImpl$DataMessageHandler.handle(NetSocketImpl.java:393) ~[vertx-core-3.7.0.jar:3.7.0]
	at io.vertx.core.streams.impl.InboundBuffer.handleEvent(InboundBuffer.java:230) ~[vertx-core-3.7.0.jar:3.7.0]
	at io.vertx.core.streams.impl.InboundBuffer.write(InboundBuffer.java:120) ~[vertx-core-3.7.0.jar:3.7.0]
	at io.vertx.core.net.impl.NetSocketImpl.handleMessage(NetSocketImpl.java:371) ~[vertx-core-3.7.0.jar:3.7.0]
	at io.vertx.core.net.impl.ConnectionBase.handleRead(ConnectionBase.java:390) ~[vertx-core-3.7.0.jar:3.7.0]
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:320) [vertx-core-3.7.0.jar:3.7.0]
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:43) [vertx-core-3.7.0.jar:3.7.0]
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:188) [vertx-core-3.7.0.jar:3.7.0]
	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:174) [vertx-core-3.7.0.jar:3.7.0]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:579) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:496) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897) [netty-common-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.30.Final.jar:4.1.30.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
```
### Expected
All items should be returned in same order with value set to null in array for which no value was found.

**Comment by vinscom** [comment]
Test https://github.com/vinscom/vertx-issue-140

**Comment by pmlopes** [comment]
Fixed on master and 3.7.1
